File: TimeTrackerPro/androidApp/src/main/java/com/appsdeviser/timetrackerpro/android/ui/core/CategoryWithColor.kt
package com.appsdeviser.timetrackerpro.android.ui.core

import androidx.compose.ui.graphics.Color
import com.appsdeviser.core_common.constants.CategoryColorCommon
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightBlue
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightGreen
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightIndigo
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightOrange
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightPink
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightPrimaryColor
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightRed
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightViolet
import com.appsdeviser.timetrackerpro.android.ui.core.theme.LightYellow

sealed class CategoryWithColor(val colorName: String = CategoryColorCommon.DEFAULT) {
    object Blue : CategoryWithColor(CategoryColorCommon.LIGHT_BLUE)
    object Green : CategoryWithColor(CategoryColorCommon.LIGHT_GREEN)
    object Red : CategoryWithColor(CategoryColorCommon.LIGHT_RED)
    object Pink : CategoryWithColor(CategoryColorCommon.LIGHT_PINK)
    object Orange : CategoryWithColor(CategoryColorCommon.LIGHT_ORANGE)
    object Violet : CategoryWithColor(CategoryColorCommon.LIGHT_VIOLET)
    object Yellow : CategoryWithColor(CategoryColorCommon.LIGHT_YELLOW)
    object Indigo : CategoryWithColor(CategoryColorCommon.LIGHT_INDIGO)
    object Default : CategoryWithColor(CategoryColorCommon.DEFAULT)

    fun getColor(): Color {
        return when(this){
            Orange -> LightOrange
            Blue -> LightBlue
            Green -> LightGreen
            Indigo -> LightIndigo
            Pink -> LightPink
            Red -> LightRed
            Violet -> LightViolet
            Yellow -> LightYellow
            Default -> LightPrimaryColor
        }
    }

    companion object {
        fun fromString(name: String): CategoryWithColor {
            return when (name) {
                CategoryColorCommon.LIGHT_ORANGE -> Orange
                CategoryColorCommon.LIGHT_BLUE -> Blue
                CategoryColorCommon.LIGHT_GREEN -> Green
                CategoryColorCommon.LIGHT_RED -> Red
                CategoryColorCommon.LIGHT_PINK -> Pink
                CategoryColorCommon.LIGHT_VIOLET -> Violet
                CategoryColorCommon.LIGHT_YELLOW -> Yellow
                CategoryColorCommon.LIGHT_INDIGO -> Indigo
                else -> Default
            }
        }
    }
}

File: TimeTrackerPro/iosApp/iosApp/core/CategoryWithColor.swift
//
//  CategoryWithColor.swift
//  iosApp
//
//  Created by AndroidLab on 2024-04-23.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

enum CategoryWithColor {
    case blue
    case green
    case red
    case pink
    case orange
    case violet
    case yellow
    case indigo
    case `default`

    var color: Color {
        switch self {
        case .blue:
            return Color.lightBlue
        case .green:
            return Color.lightGreen
        case .red:
            return Color.lightRed
        case .pink:
            return Color.lightPink
        case .orange:
            return Color.lightOrange
        case .violet:
            return Color.lightViolet
        case .yellow:
            return Color.lightYellow
        case .indigo:
            return Color.lightIndigo
        case .default:
            return Color.lightPrimaryColor
        }
    }

    static func fromString(name: String) -> CategoryWithColor {
        switch name {
        case CategoryColorCommon().LIGHT_ORANGE:
            return .orange
        case CategoryColorCommon().LIGHT_BLUE:
            return .blue
        case CategoryColorCommon().LIGHT_GREEN:
            return .green
        case CategoryColorCommon().LIGHT_RED:
            return .red
        case CategoryColorCommon().LIGHT_PINK:
            return .pink
        case CategoryColorCommon().LIGHT_VIOLET:
            return .violet
        case CategoryColorCommon().LIGHT_YELLOW:
            return .yellow
        case CategoryColorCommon().LIGHT_INDIGO:
            return .indigo
        default:
            return .default
        }
    }
}
